dialects/mysql-client.js

```diff
@@ -10,12 +10,21 @@ class MysqlClient {
       var info = url.parse(options)
       var auth = info.auth && info.auth.split(':')
       var more = info.query && querystring.parse(info.query)
+      // get port from url parser, input options, or default to 3306
+      var port = ''
```

Comment:
`var port = info.port ? info.port : options.port ? options.port : '3306'` is a cleaner option